lib/publiccloud/ec2.pm

```diff
@@ -114,14 +115,45 @@ sub upload_img {
 
     die("Create key-pair failed") unless ($self->create_keypair($self->prefix . time, 'QA_SSH_KEY.pem'));
 
+    # AMI of image to use for helper VM to create/build the image on CSP.
+    my $helper_ami_id = get_var('PUBLIC_CLOUD_EC2_UPLOAD_AMI');
+
+    # in case AMI for helper VM is not provided in job settings fallback to predefined hash
+    unless (defined($helper_ami_id)) {
+
+        # AMI is region specific also we need to use different AMI's for on-demand/BYOS uploads
+        my $ami_id_hash = {
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'us-west-1-byos' => 'ami-02b1a98531379c966',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'us-west-1' => 'ami-05c558c169cfe8d99',
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'us-west-2-byos' => 'ami-0b167c7e41ab53cb1',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'us-west-2' => 'ami-0174313b5af8423d7',
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'eu-central-1-byos' => 'ami-09740eee15629d44f',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'eu-central-1' => 'ami-09e8a19c9eda495b3',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-arm64
+            'eu-central-1-arm64' => 'ami-00f2e594abc782530'
+        };
+
+        my $ami_id_key = $self->region;
+        $ami_id_key .= '-byos'  if is_byos();
+        $ami_id_key .= '-arm64' if check_var('PUBLIC_CLOUD_ARCH', 'arm64');
+        $helper_ami_id = $ami_id_hash->{$ami_id_key} if exists($ami_id_hash->{$ami_id_key});
+    }
+
+    die('Unable to detect AMI for helper VM') unless (defined($helper_ami_id));
+
     my ($img_name)    = $file =~ /([^\/]+)$/;
     my $img_arch      = get_var('PUBLIC_CLOUD_ARCH', 'x86_64');
     my $sec_group     = get_var('PUBLIC_CLOUD_EC2_UPLOAD_SECGROUP');
     my $vpc_subnet    = get_var('PUBLIC_CLOUD_EC2_UPLOAD_VPCSUBNET');
-    my $instance_type = get_var('PUBLIC_CLOUD_EC2_UPLOAD_INSTANCE_TYPE');
-    # Used for helper VM to create/build the image on CSP. When uploading a on-demand image, this ID should point
-    # to and on-demand image. If not specified, the id gets read from ec2utils.conf file.
-    my $ami_id = get_var('PUBLIC_CLOUD_EC2_UPLOAD_AMI');
+    my $instance_type = get_var('PUBLIC_CLOUD_EC2_UPLOAD_INSTANCE_TYPE', 't2.micro');
+
+    assert_script_run('touch /root/.ec2utils.conf');
```

Comment:
maybe a comment that you add this in purpose, cause otherwise `ec2uploadimg` will fail.

Reply:
or `echo "" > /root/.ec2utils.conf` to be sure it is empty